k8s/envoy-jwt/README.md

```diff
@@ -0,0 +1,329 @@
+   - name: envoy.ext_authz
+   config:
+      grpc_service:
+         envoy_grpc:
+         cluster_name: ext-authz
+         timeout: 0.5s
+```
+
+And the corresponding cluster:
+
+``` console
+  - name: ext-authz
+    connect_timeout: 1s
+    type: strict_dns
+    http2_protocol_options: {}
+    hosts:
+      - socket_address:
+          address: 127.0.0.1
+          port_value: 9010
+```
+
+If you are curious, the complete configuration file is located at `k8s/backend/config/envoy.yaml`.
+
+## Apply the New Resource
+
+In order for the new configuration to take effect, the services need to be redeployed. Let's remove backend and frontend deployments to get them updated:
+
+```console
+   $ kubectl delete deployment backend
+   $ kubectl delete deployment frontend
+```
+
+Deploy the new resources using:
+
+```console
+   $ kubectl apply -k k8s/.
+
+   configmap/backend-envoy configured
+   configmap/be-envoy-jwt-auth-helper-config created
+   configmap/fe-envoy-jwt-auth-helper-config created
+   configmap/frontend-envoy configured
+   deployment.apps/backend configured
+   deployment.apps/frontend configured
+```
+
+## Create Registration Entries
+
+In order to fetch or validate JWT SVIDs issued by SPIRE, the `auth-helper` instances need to be authenticated on the SPIRE Server. We can achieve this by creating registration entries for each of them using the following bash script:
+
+```console
+   $ bash create-registration-entries.sh
+```
+
+Once the script is run, the list of new registration entries will be shown.
+
+```console
+   ...
+   Creating registration entry for the backend - auth-server...
+   Entry ID      : ecb140ab-50a7-4590-9fe0-d715ada67f29
+   SPIFFE ID     : spiffe://example.org/ns/default/sa/default/backend
+   Parent ID     : spiffe://example.org/ns/spire/sa/spire-agent
+   TTL           : 3600
+   Selector      : k8s:ns:default
+   Selector      : k8s:sa:default
+   Selector      : k8s:pod-label:app:backend
+   Selector      : k8s:container-name:auth-helper
+
+   Creating registration entry for the frontend - auth-server...
+   Entry ID      : 59a127fa-328c-4115-883e-5ee20b86714f
+   SPIFFE ID     : spiffe://example.org/ns/default/sa/default/frontend
+   Parent ID     : spiffe://example.org/ns/spire/sa/spire-agent
+   TTL           : 3600
+   Selector      : k8s:ns:default
+   Selector      : k8s:sa:default
+   Selector      : k8s:pod-label:app:frontend
+   Selector      : k8s:container-name:auth-helper
+   ...
+```
+
+Note that the selectors for new services points to the `auth-helper` container: `k8s:container-name:auth-helper`. This is necessary to authenticate the service into SPIRE so then it can fetch/validate the JWT SVIDs configured as an authentication header for every request.
+
+Intentionally, there is no registration entry for the frontend-2 service. It will be added later to demonstrate that requests are denied by the external authorization filter when a JWT-SVID is not present in the request header.
```

Comment:
```suggestion
Intentionally, there is no registration entry for the `frontend-2` service. It will be added later to demonstrate that requests are denied by the external authorization filter when a JWT-SVID is not present in the request header.
```